packages/aws-lambda/lambdas/layer-version-api/index.js

```diff
@@ -107,11 +92,17 @@ async function handleLayerRequest(event, layerName) {
 
   console.log(`No cached result for ${cacheKey}.`);
   const data = await listLayerVersions(region, layerName);
+
   if (!data) {
     return respond500('No result from AWS ListLayerVersion operation.');
   }
+
   if (!data.LayerVersions || !data.LayerVersions[0] || !data.LayerVersions[0].Version) {
-    return respond500('Unexpected result from AWS ListLayerVersion operation.');
+    if (Array.isArray(data.LayerVersions) && data.LayerVersions.length === 0) {
+      return respond500('No layer version found.');
```

Comment:
_nitpick:_ this could probably be a 404 as well -- since we did not find a layer in the requested region (instead of a 500).

Reply:
Yeah was too lazy to create a new fn for that. But will doo now

Reply:
Fixed